From 02789309edfef3309658532e73e0119528bece4f Mon Sep 17 00:00:00 2001
From: Michal Vasilek <michal.vasilek@nic.cz>
Date: Fri, 1 Oct 2021 17:44:38 +0200
Subject: [PATCH] mvebu, mpc85xx: enable BFQ

BFQ is an IO scheduler which enables higher IO throughput on hard drives
on sequential workloads which is especially nice for NASes.

It also BFQ hierarchical scheduling support [2]

[2] https://cateee.net/lkddb/web-lkddb/BFQ_GROUP_IOSCHED.html

Signed-off-by: Michal Vasilek <michal.vasilek@nic.cz>
---
 target/linux/mpc85xx/config-5.10 | 2 ++
 target/linux/mvebu/config-5.10   | 2 ++
 2 files changed, 2 insertions(+)

diff --git a/target/linux/mpc85xx/config-5.10 b/target/linux/mpc85xx/config-5.10
index e68ed34646..c91a520c9c 100644
--- a/target/linux/mpc85xx/config-5.10
+++ b/target/linux/mpc85xx/config-5.10
@@ -113,6 +113,8 @@ CONFIG_I2C_BOARDINFO=y
 CONFIG_I2C_MPC=y
 CONFIG_ILLEGAL_POINTER_VALUE=0
 CONFIG_INITRAMFS_SOURCE=""
+CONFIG_IOSCHED_BFQ=y
+CONFIG_BFQ_GROUP_IOSCHED=y
 CONFIG_IRQCHIP=y
 CONFIG_IRQ_DOMAIN=y
 CONFIG_IRQ_FORCED_THREADING=y
diff --git a/target/linux/mvebu/config-5.10 b/target/linux/mvebu/config-5.10
index d16564e5d7..51ef37fadf 100644
--- a/target/linux/mvebu/config-5.10
+++ b/target/linux/mvebu/config-5.10
@@ -217,6 +217,8 @@ CONFIG_I2C_CHARDEV=y
 CONFIG_I2C_MV64XXX=y
 # CONFIG_I2C_PXA is not set
 CONFIG_INITRAMFS_SOURCE=""
+CONFIG_IOSCHED_BFQ=y
+CONFIG_BFQ_GROUP_IOSCHED=y
 CONFIG_IRQCHIP=y
 CONFIG_IRQ_DOMAIN=y
 CONFIG_IRQ_DOMAIN_HIERARCHY=y
